VaultFS

BrianHicks opened the pull request.

BrianHicks committed:
vaultfs: add

BrianHicks committed:
README: add vaultfs

siddharthist commented on vault/vaultfs/vaultfs-docker.service:
```diff
@@ -0,0 +1,15 @@
+[Unit]
+Description=Docker volume plugin for vaultfs
+Documentation={{.URL}}
+After=network-online.target
```
Should this also be after `docker.service`?

BrianHicks replied:
Yes, probably

Commit:
Added Vaultfs Support

Added Vaultfs support by creating a new service `vaultfs.service` that
mounts `vault` at /var/vault.

Commit:
Merge branch 'feature/vaultfs' of https://github.com/asteris-llc/mantl-packaging into feature/vaultfs

Commit:
Merge branch 'master' of https://github.com/asteris-llc/mantl-packaging into feature/vaultfs

Commit:
Merge branch 'master' of https://github.com/asteris-llc/mantl-packaging into feature/vaultfs

Commit:
Merge branch 'feature/vaultfs' of https://github.com/asteris-llc/mantl-packaging into feature/vaultfs

Commit:
Merge branch 'feature/vaultfs' of https://github.com/asteris-llc/mantl-packaging into feature/vaultfs

Dacode45 commented:
Added Vaultfs support by creating a new service `vaultfs.service` that
mounts `vault` at /var/vault.

BrianHicks commented on README.md:
```diff
@@ -26,6 +26,7 @@ for building generic Mantl utilities.
 		- [Vault Packages](#vault-packages)
 			- [vault](#vault)
 			- [vault-mantl](#vault-mantl)
+			- [vault-packages](#vaultfs)
```
the name doesn't match the header here

Dacode45 replied:
Fixed.

BrianHicks commented on vault/vault-mantl/spec.yml:
```diff
@@ -15,13 +15,13 @@ depends:
 targets:
   - src: "{{.SpecRoot}}/vault-unseal.sh"
-    dest: /usr/local/bin
+    dest: /usr/local/bin/vault-unseal.sh
```
why did you change these?

Dacode45 replied:
I got a weird error where the package wouldn't build unless this was set explicitly.

Commit:
Merge branch 'feature/vaultfs' of https://github.com/asteris-llc/mantl-packaging into feature/vaultfs

BrianHicks commented on vault/vaultfs/vaultfs-bootstrap.sh:
```diff
@@ -0,0 +1,12 @@
+  VAULT_TOKEN=${SECURE_TOKEN_PATH}
+fi
+
+vaultfs mount --address=$VAULT_ADDRESS -t ${VAULT_TOKEN} ${VAULT_MOUNTPOINT}
```
this will leave the daemon running. Is that what you want?

Dacode45 replied:
Fixed. Now runs as a forked process.

Commit:
Vaultfs service now runs as a daemon.

Commit:
Vaultfs service now runs as a daemon.

Commit:
Merge branch 'feature/vaultfs' of https://github.com/asteris-llc/mantl-packaging into feature/vaultfs

siddharthist commented:
@Dacode45 Looks like something strange has happened to the git history here...

Commit:
Changed line endings from window style to unix style in script files.

Dacode45 commented:
The commit history looks weird because I did a last minute amend to a commit to add a Readme change. Everything looks okay though.

BrianHicks commented:
This has drifted very far out of date. I'm happy to help transfer context on a rewrite, but am going to close it for now as I'm not able to be responsible for rewriting/merging it at this point.

Theaxiom commented:
Thanks @BrianHicks -- if you want to be added to Mantl org, let me know.
